### DISCUSSION TITLE
How to enable CSP(Content Security Policy) in SSR mode

### DISCUSSION THREAD

**Discussion by fengjac** [quasarframework/quasar#17511] [CLI - SSR mode]
I try to enable CSP in production environment.
```javascript
/src-ssr/server.ts
export const create = ssrCreate((/* { ... } */) => {
  ...
  if (process.env.PROD) {
    // add for security reasons
    app.use(helmet({ contentSecurityPolicy: true }))
  }
  ...
  return app
})
```
But it was failed to run in browser and reported errors in browser console like this:
![image](https://github.com/user-attachments/assets/4b0b8c31-9f9d-4e7c-b34a-eda23bc5965e)

Is anybody could give me a hand? Thanks a lot!

**Comment by rstoenescu** [comment]
Edit your /index.html (or /src/index.template.html) and add the necessary meta tag. Or use Quasar's Meta plugin to add it dynamically. There's a number of ways this can be done.

**Comment by yusufkandemir** [comment]
Check the contents of the said file and search for any traces of `eval(...)`. I would recommend building with `--debug` flag to make it easier. After you find it, you will have to replace the related library to satisfy the CSP.
